worlds/kh1/__init__.py

```diff
@@ -0,0 +1,287 @@
+
+
+class KH1World(World):
+    """
+    Kingdom Hearts is an action RPG following Sora on his journey 
+    through many worlds to find Riku and Kairi.
+    """
+    game = "Kingdom Hearts"
+    options_dataclass = KH1Options
+    options: KH1Options
+    topology_present = True
+    web = KH1Web()
+
+    item_name_to_id = {name: data.code for name, data in item_table.items()}
+    location_name_to_id = {name: data.code for name, data in location_table.items()}
+    item_name_groups = item_name_groups
+    location_name_groups = location_name_groups
+    fillers = {}
+    fillers.update(get_items_by_category("Item"))
+    fillers.update(get_items_by_category("Camping"))
+    fillers.update(get_items_by_category("Stat Ups"))
+
+    def create_items(self):
+        self.place_predetermined_items()
+        # Handle starting worlds
+        starting_worlds = []
+        if self.options.starting_worlds > 0:
+            possible_starting_worlds = ["Wonderland", "Olympus Coliseum", "Deep Jungle", "Agrabah", "Monstro", "Halloween Town", "Neverland", "Hollow Bastion"]
+            if self.options.atlantica:
+                possible_starting_worlds.append("Atlantica")
+            if self.options.end_of_the_world_unlock == "item":
+                possible_starting_worlds.append("End of the World")
+            starting_worlds = self.random.sample(possible_starting_worlds, min(self.options.starting_worlds.value, len(possible_starting_worlds)))
+            for starting_world in starting_worlds:
+                self.multiworld.push_precollected(self.create_item(starting_world))
+        
+        item_pool: List[KH1Item] = []
+        possible_level_up_item_pool = []
+        level_up_item_pool = []
+        
+        # Calculate Level Up Items
+        # Fill pool with mandatory items
+        for _ in range(self.options.item_slot_increase):
+            level_up_item_pool.append("Item Slot Increase")
+        for _ in range(self.options.accessory_slot_increase):
+            level_up_item_pool.append("Accessory Slot Increase")
+
+        # Create other pool
+        for _ in range(self.options.strength_increase):
+            possible_level_up_item_pool.append("Strength Increase")
+        for _ in range(self.options.defense_increase):
+            possible_level_up_item_pool.append("Defense Increase")
+        for _ in range(self.options.hp_increase):
+            possible_level_up_item_pool.append("Max HP Increase")
+        for _ in range(self.options.mp_increase):
+            possible_level_up_item_pool.append("Max MP Increase")
+        for _ in range(self.options.ap_increase):
+            possible_level_up_item_pool.append("Max AP Increase")
+
+        # Fill remaining pool with items from other pool
+        while len(level_up_item_pool) < 100 and len(possible_level_up_item_pool) > 0:
+            level_up_item_pool.append(possible_level_up_item_pool.pop(self.random.randrange(len(possible_level_up_item_pool))))
```

Comment:
Probably just shuffle `possible_level_up_item_pool` and use `pop` normally

```
self.random.shuffle(level_up_item_pool)
while len(level_up_item_pool) < 100 and len(possible_level_up_item_pool) > 0:
    level_up_item_pool.append(possible_level_up_item_pool.pop())
```

In fact, wasn't it like that before? I was ok with that. If you thought I had an issue with it, then my bad for being unclear :)